src/api/app/models/package.rb

```diff
@@ -1365,6 +1367,39 @@ def save_file(opt = {})
     sources_changed(wait_for_update: ['_aggregate', '_constraints', '_link', '_service', '_patchinfo', '_channel'].include?(opt[:filename]))
   end
 
+  def save_files(opt = {})
```

Comment:
I would prefer to see this kind of code - that has nothing to do with persistence - happening in another level, i.e a service object, being called in the controller. IMO `User.session` should leak as less as possible to the model layer..

Reply:
Yeah, it would probably be the best to do that in the controller, which would allow to also commit `_service` in the same filelist if correctly implemented.